zkevm-circuits/src/evm_circuit/util/math_gadget/abs_word.rs

```diff
@@ -120,72 +120,65 @@ mod tests {
 
         fn assign_gadget_container(
             &self,
-            input_words: &[Word],
+            witness_words: &[Word],
             region: &mut CachedRegion<'_, '_, F>,
         ) -> Result<(), Error> {
             let offset = 0;
-            let x = input_words[0];
-            let x_abs = input_words[1];
+            let x = witness_words[0];
+            let x_abs = witness_words[1];
             self.absword_gadget.assign(region, offset, x, x_abs)?;
 
             Ok(())
         }
     }
 
+    macro_rules! try_test_abs_x {
+        ($x:expr; is neg, abs eq $x_abs:expr, $expect_success:expr) => {{
+            try_test!(AbsWordGadgetContainer<Fr, true>, [$x, $x_abs], $expect_success)
+        }};
+        ($x:expr; is not neg, abs eq $x_abs:expr, $expect_success:expr) => {{
+            try_test!(AbsWordGadgetContainer<Fr, false>, [$x, $x_abs], $expect_success)
+        }};
+    }
+
```

Comment:
- Would just call the macro`try_test` in all files to make it as short as possible, I don't think there can be any confusion.
- The `is neg`, `abs eq` etc... is a bit strange to me, would just keep the syntax easy with normal parameters.

Something like `try_test!(Word::from(0), Word::from(0), true);` seems easy to read to me.

Reply:
Sure, a plain macro calling is enough I think. I just tried more interesting features of rust macro, now I know why so many DSLs are created/based on rust.